staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -219,8 +311,110 @@ func (q *delayingType) AddAfter(item interface{}, duration time.Duration) {
 	select {
 	case <-q.stopCh:
 		// unblock if ShutDown() is called
-	case q.waitingForAddCh <- &waitFor{data: item, readyAt: q.clock.Now().Add(duration)}:
+	case q.waitingForAddCh <- &waitFor{
+		data:    item,
+		readyAt: q.clock.Now().Add(duration),
+		options: ExpandedDelayingOptions{
+			PermitActiveAndWaiting: true,
```

Comment:
I know it is the first iota value, but can we explicitly pass `TakeShorter` to emphasize what we want to achieve here?

Reply:
Have done, thanks.